File: 백준/Silver/1475. 방 번호/방 번호.java
import java.util.*;
import java.io.*;


class Main
{
    public static void main (String[] args) throws IOException {
        Scanner in = new Scanner(System.in);
        String s = in.nextLine();
        int[] arr = new int[s.length()];
        for (int i = 0; i < s.length(); i++) arr[i] = s.charAt(i) - '0';

        int res = 0, tmp = 0;
        int[] count = new int[10];

        for (int i = 0; i < s.length(); i++){
            tmp = arr[i];
            //원소가 없을때
            if (count[tmp] == 0){
                if (tmp == 6 || tmp == 9){
                    if (count[6] == 0 && count[9] == 0){
                        for (int j = 0; j < 10; j++) count[j]++;
                        count[tmp]--;
                        res++;
                    }
                    else if (count[6] > 0) count[6]--;
                    else if (count[9] > 0) count[9]--;
                }

                else {
                    for (int j = 0; j < 10; j++) count[j]++;
                    count[tmp]--;
                    res++;
                }
            }
            // 원소가 잇으면 갯수를 하나 빼주기
            else count[tmp]--;
        }
        System.out.println(res);
    }
}

File: 백준/Silver/1475. 방 번호/방 번호.cc
#include <bits/stdc++.h>
using namespace std;



int main(void) {

    ios::sync_with_stdio(0);
    cin.tie(0);

    int arr[10] = {};
    int res = 0;

    string s;
    cin >> s;

    for (int i = 0; i < s.size(); i++) {
        int tmp = s[i] - '0';
        if (arr[tmp] == 0) {
            if (tmp == 6) {
                if (arr[9] > 0) {
                    arr[9]--;
                    continue;
                }
            }
            else if (tmp == 9) {
                if (arr[6] > 0) {
                    arr[6]--;
                    continue;
                }
            }

            res++;
            for (int j = 0; j < 10; j++) arr[j] = arr[j]+1;
            arr[tmp]--;
        }
        else arr[tmp]--;
    }

    cout << res;
}